deploy/charts/preflight/templates/agent_installation.yaml

```diff
@@ -0,0 +1,214 @@
+kind: ClusterRole
+metadata:
+  name: jetstack-secure-agent-googlecas-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+rules:
+- apiGroups: ["cas-issuer.jetstack.io"]
+  resources:
+  - googlecasissuers
+  - googlecasclusterissuers
+  verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: jetstack-secure-agent-get-webhooks
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+rules:
+- apiGroups: ["admissionregistration.k8s.io"]
+  resources:
+  - validatingwebhookconfigurations
+  - mutatingwebhookconfigurations
+  verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-secret-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-secret-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-cert-manager-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-cert-manager-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-googlecas-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-googlecas-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-get-webhooks
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-get-webhooks
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-node-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-node-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-cluster-viewer
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: view
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
```

Comment:
Since the Marketplace release does not allow us to create the RBAC roles, rolebindings and service accounts in the Helm chart, could we have a flag here to disable the creation of service accounts and RBAC roles and rolebindings?

e.g. like [in cert-manager](https://github.com/jetstack/cert-manager/blob/35febb171706826f27d71af466c624c25733c135/deploy/charts/cert-manager/templates/rbac.yaml#L1):

```yaml
{{- if .Values.global.rbac.create -}}

{{ end }}
```